lib/xain/helpers.ex

```diff
@@ -37,6 +38,18 @@ defmodule Xain.Helpers do
   def pop!([item | rest]) do
     {rest, item |> Enum.reverse}
   end
+
+  def ensure_valid_contents(contents, _) when is_binary(contents) or is_list(contents) do
+    contents
+  end
+  def ensure_valid_contents(contents, tag_name) when is_number(contents) or is_atom(contents) do
+    Logger.debug "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    to_string(contents)
+  end
+  def ensure_valid_contents(contents, tag_name) do
+    Logger.warn "#{tag_name} has been called as #{tag_name}(#{inspect(contents)}, ...), but the first argument supposed to be a binary"
+    inspect(contents)
```

Comment:
Normally, stuff that does not support to_string will raise an exception. I think I would prefer to keep that behaviour consistent. Furthermore, what about the case where they may implement the `String.Chars` protocol for a custom type? 

If we have just Logger.debug for anything non-binary and then call `to_string` well raise an exception for anything that does not support String.Chars. We can leave it up to the application to catch that error if they want to handle it. So, in ex_admin, we could try catching the error and raise a more detailed error.